packages/mdc-dialog/util.ts

```diff
@@ -24,26 +24,24 @@
 import * as createFocusTrap from 'focus-trap';
 import {FocusTrapFactory} from './types';
 
-function createFocusTrapInstance(
+export function createFocusTrapInstance(
     surfaceEl: HTMLElement,
     focusTrapFactory: FocusTrapFactory = createFocusTrap as unknown as FocusTrapFactory,
-    initialFocusEl: createFocusTrap.FocusTarget | null,
+    initialFocusEl?: createFocusTrap.FocusTarget | null,
 ): createFocusTrap.FocusTrap {
-  return focusTrapFactory(surfaceEl, ({
-    clickOutsideDeactivates: true, // Allow handling of scrim clicks
-    escapeDeactivates: false, // Dialog foundation handles escape key
-    initialFocus: initialFocusEl,
-  } as createFocusTrap.Options));
+  return focusTrapFactory(surfaceEl, {
+    clickOutsideDeactivates: true, // Allow handling of scrim clicks.
+    escapeDeactivates: false, // Foundation handles ESC key.
+    initialFocus: initialFocusEl || undefined,
```

Comment:
I was getting a compiler error with Drawer's `createFocusTrapInstance()`, so I compared it to Dialog and noticed some differences. I figured I would make them both consistent as long as I was here. Same with the unit tests below.

It wasn't immediately possible to avoid passing `null` to `initialFocusEl`, because `mdc-dialog/index.ts` has this:

```ts
class MDCDialog extends MDCComponent<MDCDialogFoundation> {
  private initialFocusEl_!: HTMLElement | null; // assigned in initialize()

  initialize(
      focusTrapFactory: FocusTrapFactory = createFocusTrap as unknown as FocusTrapFactory,
      initialFocusEl: Element | null = null) {
    this.initialFocusEl_ = initialFocusEl as HTMLElement | null;
  }

  initialSyncWithDOM() {
    this.focusTrap_ = util.createFocusTrapInstance(this.container_, this.focusTrapFactory_, this.initialFocusEl_);
  }
}
```

However, I can just make `initialFocusEl` optional instead of nullable.

Reply:
Made `initialFocusEl` optional instead of nullable.